src/data-sources/collection/dapp-radar/dapp.datasource.ts

```diff
@@ -0,0 +1,175 @@
+import { BadRequestException } from '@nestjs/common';
+import axios, { AxiosHeaders, AxiosInstance } from 'axios';
+
+import { Protocol, METRICS } from '../../../entities/protocol.entity';
+import { Units } from '../../../entities/entity';
+import { USD } from '../../../common/currecies';
+import { Meta } from '../../abstract.data-source';
+import { addLogging } from '../../../common/http';
+import { URL_REGEXP } from '../../../common/regexp';
+import { AbstractDappDataSource } from '../../abstract.dapp.data-source';
+
+type Config = {
+  key: string;
+  endpoint: string;
+};
+
+const RANGES = ['24h', '7d', '30d'];
+type Range = (typeof RANGES)[number];
+
+type Params = {
+  dappId: number;
+  range: Range;
+};
+
+type DappRadarApp = {
+  dappId: number;
+  name: string;
+  logo: string | null;
+  chains: string[];
+  metrics: Record<string, number>;
+};
+
+const scalarMetrics = ['transactions', 'uaw'];
+
+export class DappRadarDappDatasource extends AbstractDappDataSource<
+  Config,
+  Params
+> {
+  private httpClient: AxiosInstance;
+
+  public static getName(): string {
+    return `DappRadar DApps`;
```

Comment:
what is the difference between DappRadarDappDatasource and DappRadarProtocolDatasource in the eyes of the user? like, how one should select one of them?